src/main/java/com/hubspot/imap/client/ImapClient.java

```diff
@@ -458,7 +458,18 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
   public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
     if (evt instanceof IdleStateEvent) {
       if (!connectionShutdown.get()) {
-        noop();
+        CompletableFuture<NoopResponse> noopResponse = noop();
+        noopResponse.whenComplete((resp, throwable) -> {
+          if (channel.isOpen()) {
+            if (throwable != null) {
+              logger.warn("Exception from NOOP", throwable);
+              //closeNow();
```

Comment:
why is this commented out?

Reply:
I wanted to see how often this warn log pops up and make sure this isn't something that happens regularly even to healthy inboxes. And also make sure that the log fits into the timeline of the problem that I am trying to fix.